internal/pkg/client/github/fetch.go

```diff
@@ -0,0 +1,87 @@
+// -----------------------------------------------------------------------------
+// SPDX-FileCopyrightText: Copyright © 2024 bomctl a Series of LF Projects, LLC
+// SPDX-FileName: internal/pkg/client/github/fetch.go
+// SPDX-FileType: SOURCE
+// SPDX-License-Identifier: Apache-2.0
+// -----------------------------------------------------------------------------
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// -----------------------------------------------------------------------------
+
+package github
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/google/go-github/v66/github"
+	"golang.org/x/oauth2"
+
+	"github.com/bomctl/bomctl/internal/pkg/netutil"
+	"github.com/bomctl/bomctl/internal/pkg/options"
+)
+
+func (client *Client) Fetch(fetchURL string, opts *options.FetchOptions) ([]byte, error) {
+	url := client.Parse(fetchURL)
+	ctx := context.Background()
+	auth := netutil.NewBasicAuth(url.Username, url.Password)
+
+	if opts.UseNetRC {
+		if err := auth.UseNetRC(url.Hostname); err != nil {
+			return nil, fmt.Errorf("failed to set auth: %w", err)
+		}
+	}
+
+	ts := oauth2.StaticTokenSource(&oauth2.Token{AccessToken: auth.Password})
+	tc := oauth2.NewClient(ctx, ts)
+	client.ghClient = *github.NewClient(tc)
+
+	repoURL := strings.Split(url.Path, "/")
+	owner := repoURL[0]
+	repo := repoURL[1] + "/dependency-graph/sbom"
+	u := fmt.Sprintf("repos/%s/%s", owner, repo)
```

Comment:
I'm surprised the linter is letting these slide. `ts`, `tc`, and `u` are too short for variable names. Can you rename them to `tokenSource`, `tokenClient`/`oauthClient`, and `apiPath`/`endpoint` or something like that?